### PR TITLE
server: combine default balancer and others.

### PR DESCRIPTION
**Pull request by siddontang**
@qiuyesuifeng @disksing @shenli

### CODE REVIEW

File: server/cluster_worker.go [hunk 1]
```diff
@@ -34,26 +35,37 @@ func (c *raftCluster) handleDefaultBalancer(region *metapb.Region, leader *metap
...
+
+		if balanceOperator == nil {
+			return nil, nil
+		}
 	}
 
 	ret, res, err := balanceOperator.Do(region, leader)
-	if err != nil {
-		log.Errorf("do balance for region %d failed %s", regionID, err)
+	if ret || err != nil {
```
Comment (qiuyesuifeng) [inline server/cluster_worker.go]: I  suggest we separate checking error and ret value.

Comment (siddontang) [inline server/cluster_worker.go]: No, here these cases have same handling later, we can handle them together.

Comment (qiuyesuifeng) [inline server/cluster_worker.go]: It is a code style for us now.

File: server/cluster_worker.go [hunk 2]
```diff
@@ -34,26 +35,37 @@ func (c *raftCluster) handleDefaultBalancer(region *metapb.Region, leader *metap
...
+		// Do finished or failed, remove it.
 		c.balanceWorker.removeBalanceOperator(regionID)
-	}
 
-	if ret {
-		// Do finished, remove it.
-		c.balanceWorker.removeBalanceOperator(regionID)
+		if err != nil {
```
Comment (qiuyesuifeng) [inline server/cluster_worker.go]: Alignment.

**Comment by qiuyesuifeng** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @shenli @disksing

**Pull request merged**